src/DataMapper/DataMapper.php

```diff
@@ -635,13 +635,14 @@ private function calculateChanges(array &$changes, array $newData, array $oldDat
             }
         }
 
+        $isArrayList = array_is_list($oldData);
         foreach ($oldData as $k => $v) { // data that used to exist, but now doesn't
             if (!isset($newData[$k])) { // removed field
-                if (is_integer($k)) {
+                $changes['$unset']["{$keyfix}{$k}"] = '';
+
+                if ($isArrayList) { // is integer and not associative array
```

Comment:
I approved, but looking closer, before it checked if the index k was an integer `is_integer($k)` and now it's checking if the array is not an associative array, but this `array_is_list` function returns false if the first index starts at 1. Is it right? It's a breaking change, right?